File: =EXAMTIME/1_TradeDesk/PushSwichFile  Commands/p.java
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;

public class p {
    public static String processLogs(List<String> logs) {
        Map<String, Set<String>> branches = new HashMap<>();
        String currentBranch = "main";

        for (String log : logs) {
            String[] parts = log.split("\\s");
            String action = parts[0];

            if (action.equals("switch")) {
                currentBranch = parts[1];
            } else if (action.equals("push")) {
                String fileName = parts[1];

                branches.computeIfAbsent(currentBranch, k -> new HashSet<>()).add(fileName);
            }
        }

        String maxBranch = "";
        int maxCount = 0;

        for (Map.Entry<String, Set<String>> entry : branches.entrySet()) {
            int branchSize = entry.getValue().size();
            if (branchSize > maxCount) {
                maxCount = branchSize;
                maxBranch = entry.getKey();
            }
        }

        return maxBranch;
    }

    public static void main(String[] args) {
        List<String> inputList = List.of("switch A", "push 1", "push 2", "switch B", "push 3", "push 4", "push 5", "switch A", "push 6");
        String result = processLogs(inputList);
        System.out.println(result);
    }
}

File: =EXAMTIME/1_TradeDesk/PushSwichFile  Commands/p.py
def solve(logs): 
    branches = {}
    current_branch = "main"

    for log in logs:
        action, *args = log.split()

        if action == "switch":
            current_branch = args[0]
        elif action == "push":
            branches.setdefault(current_branch, set()).add(args[0])

    max_branch = max(branches, key=lambda branch: len(branches[branch]), default="main")
    return max_branch

# Example usage:
# this and WQinner Both tested Write any 
input_list = ["switch branch1",
               "push file1",
                 "push file2",
                 "push file1",
                   "switch branch2",
                     "switch issue2", 
                     "push file1",
                       "push file2", 
                       "push file3"]
result = solve(input_list)
print(result)
